hf/run_dpo_no_trainer.py

```diff
@@ -105,34 +109,37 @@ def eval_fn(model, ref_model, eval_device_loader, config, step, tracker):
     )
 
 
-@hydra.main(version_base=None, config_path="config", config_name="config")
+def hydra_decorator(config_path, config_name):
+    def decorator(func):
+        if not USE_CUDA:
+            return hydra.main(version_base=None, config_path=config_path, config_name=config_name)(func)
+    return decorator
+
+@hydra_decorator(config_path="config", config_name="config")
 def main(config: DictConfig):
     OmegaConf.resolve(config)
     set_seed(config.seed)
     logger.info("\n\n************** Experiment configuration ***********")
     logger.info(OmegaConf.to_yaml(config))
 
-    config_path = os.path.join(config.run_dir, "config.yaml")
-    with get_file(config_path, "w") as f:
-        OmegaConf.save(config, f)
+    if not USE_CUDA:
+        config_path = os.path.join(config.run_dir, "config.yaml")
+        with get_file(config_path, "w") as f:
+            OmegaConf.save(config, f)
 
-    accelerator = Accelerator(log_with="tensorboard", project_dir=config.run_dir)
-    tracker = Tracker(config, accelerator, logger)
-    setup_xla(config)
+        accelerator = Accelerator(log_with="tensorboard", project_dir=config.run_dir)
+        tracker = Tracker(config, accelerator, logger)
+        setup_xla(config)
 
-    tokenizer = AutoTokenizer.from_pretrained(config.model.name_or_path)
-    model, ref_model, optimizer = setup_model_optimizer(config)
+        tokenizer = AutoTokenizer.from_pretrained(config.model.name_or_path)
+        model, ref_model, optimizer, scheduler = setup_model_optimizer(config)
 
     if tokenizer.vocab_size != model.config.vocab_size:
         logger.warning(
             f"Found mismatch between {tokenizer.vocab_size=} and {model.config.vocab_size}"
         )
-    train_device_loader, eval_device_loader = get_input_pipeline(config, tokenizer)
+    train_device_loader, eval_device_loader, train_ds, _ = get_input_pipeline(config, tokenizer)
```

Comment:
what's the meaning of this change and why we need to return `train_ds`?
Shall we change `get_input_pipeline` in `hf/input_pipeline_tpu.py` accordingly?

Reply:
This is for using the `init_using_ptl`, for https://github.com/NVIDIA/NeMo-Aligner/blob/9ddefe6218cf70891d46d8decdca08e692a19bed/nemo_aligner/utils/train_script_utils.py#L99

Reply:
Thank you!

Could you change get_input_pipeline in `hf/input_pipeline_tpu.py` accordingly?

Reply:
Added two None return values in `hf/input_pipeline_tpu.py`